test/extensions/filters/network/local_ratelimit/local_ratelimit_fuzz_test.cc

```diff
@@ -0,0 +1,99 @@
+#include "envoy/extensions/filters/network/local_ratelimit/v3/local_rate_limit.pb.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/stats/isolated_store_impl.h"
+
+#include "extensions/filters/network/local_ratelimit/local_ratelimit.h"
+
+#include "test/extensions/filters/network/local_ratelimit/local_ratelimit_fuzz.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::NiceMock;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace LocalRateLimitFilter {
+struct ActiveFilter {
+  ActiveFilter(const ConfigSharedPtr& config) : filter_(config) {
+    filter_.initializeReadFilterCallbacks(read_filter_callbacks_);
+  }
+
+  NiceMock<Network::MockReadFilterCallbacks> read_filter_callbacks_;
+  Filter filter_;
+};
+
+DEFINE_PROTO_FUZZER(
+    const envoy::extensions::filters::network::local_ratelimit::LocalRateLimitTestCase& input) {
+
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  } catch (const ProtobufMessage::DeprecatedProtoFieldException& e) {
+    ENVOY_LOG_MISC(debug, "DeprecatedProtoFieldException: {}", e.what());
+    return;
+  }
+  if (input.config().token_bucket().fill_interval().nanos() < 0) {
+    // TODO:
+    // protoc-gen-validate has an issue on type "Duration" which may generate interval with seconds
```

Comment:
I would also file an issue about this in protoc-gen-validate.

Reply:
> I would also file an issue about this in protoc-gen-validate.

I will do that. Do you think I should file the issue in this github repo or email someone in PGV group?

Reply:
github repo is good! you can tag me or akonradi

Reply:
Can you also add a reference to the bug in the comment here? Thanks.

Reply:
> Can you also add a reference to the bug in the comment here? Thanks.

Added.
(https://github.com/envoyproxy/protoc-gen-validate/issues/348)